packages/footer/hooks/TrackingContext.ts

```diff
@@ -0,0 +1,41 @@
+import { createContext, useContext } from 'react';
+
+/**
+ * @see https://matomo.org/docs/event-tracking/
+ * @see https://developer.matomo.org/api-reference/tracking-javascript
+ */
+export type TrackEventMatomo = {
+   /**
+    * Describes the type of events you want to track.
+    * For example, Link Clicks, Videos, Outbound Links, and Form Events.
+    */
+   eventCategory: string;
+   /**
+    * The specific action that is taken.
+    * For example, with a Video category, you might have a Play, Pause and Complete action.
+    */
+   eventAction: string;
+   /**
+    * Usually the title of the element that is being interacted with, to aid with analysis.
+    * For example, it could be the name of a Video that was played or the specific
+    * form that is being submitted.
+    */
+   eventName?: string;
+   /**
+    * A numeric value and is often added dynamically. It could be the cost of a
+    * product that is added to a cart, or the completion percentage of a video.
+    */
+   eventValue?: number;
+};
+
+type Track = (trackData: TrackEventMatomo) => void;
+type TrackContext = {
+   trackClick: Track;
+};
+
+const defaultTracker: TrackContext = {
+   trackClick: (_trackData: TrackEventMatomo) => {},
+};
+
+export const TrackingContext = createContext(defaultTracker);
+export const ClickTracker = TrackingContext.Provider;
```

Comment:
Minor naming question:

I'm curious, do we ever anticipate tracking events that are not the result of a click? Like perhaps a specific kind of page view, or a keystroke? Perhaps the names `trackEvent` or `EventTracker` are a little ambiguous though.

Reply:
Renaming this to `EventTracker` sounds reasonable. The `useTrackedOnClick` hook is already named appropriately, and I can see this being used for other events.

Reply:
[bb94039](https://github.com/iFixit/react-commerce/pull/1009/commits/bb94039321b97f744de0d5a8c3f64b4ac4ddf612)